src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1369,6 +1428,7 @@ protected MembersAndInitializers GetMembersAndInitializers()
 
             AddDeclarationDiagnostics(diagnostics);
             diagnostics.Free();
+            Interlocked.Exchange(ref _lazyDeclaredMembersAndInitializers, null);
```

Comment:
>Interlocked.Exchange(ref _lazyDeclaredMembersAndInitializers, null); [](start = 12, length = 68)

Is there a real need for an interlocked exchange here?

Reply:
Looks like it's not and all writes are volatile. Thanks

---
In reply to: [563841575](https://github.com/dotnet/roslyn/pull/50650#discussion_r563841575) [](ancestors = 563841575)